### ISSUE TITLE
Activation function clarification

### ISSUE THREAD

**Issue by nikogamulin** [qubvel/segmentation_models.pytorch#288]
Hi,

I have used the library to train DeepLabV3+. Checking some notebooks on Kaggle, I am not sure whether there is an error or I don't understand the usage of the activation function with respect to the number of outputs correctly.

In case there is a binary output (e.g. ['background', 'polyp'] - as used in [this notebook](https://www.kaggle.com/balraj98/polyp-segmentation-in-colonoscopy-frames-deeplab)), sigmoid is appropriate activation function.
In case of multiple outputs (e.g. ['urban_land', 'agriculture_land', 'rangeland', 'forest_land', 'water', 'barren_land', 'unknown'] in Land Cover Classification challenge on Kaggle), softmax2d is the right activation function, I guess. In [this notebook](https://www.kaggle.com/balraj98/deepglobe-land-cover-classification-deeplabv3), however, sigmoid has been used for activation and therefore I am not sure whether there is some reason in this specific case that the model learns correctly or is it an error.

I would appreciate if anyone commented the usage of activation functions depending on the task and clarified the case of sigmoid in multiclass problem - is it an error or could it output the correct values.

# Update

So I guess both sigmoid and softmax2d is correct for multiclass problem. In case of sigmoid, the output is multi-label (number of outputs is the number of classes) and in case of softmax2d, there is a single output and its value corresponds to a class.
